placeholder/src/main/java/com/google/accompanist/placeholder/Placeholder.kt

```diff
@@ -0,0 +1,150 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.accompanist.placeholder
+
+import androidx.compose.animation.core.animateFloat
+import androidx.compose.animation.core.rememberInfiniteTransition
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.draw.DrawModifier
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Outline
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.graphics.drawOutline
+import androidx.compose.ui.graphics.drawscope.ContentDrawScope
+import androidx.compose.ui.platform.debugInspectorInfo
+import androidx.compose.ui.unit.LayoutDirection
+
+/**
+ * If [visible], draws [shape] with a solid [color] instead of content.
+ *
+ * @sample com.google.accompanist.sample.placeholder.PlaceholderBasicSample
+ *
+ * @param visible defines whether the placeholder should be visible
+ * @param color color to paint placeholder with
+ * @param shape desired shape of the placeholder
+ */
+fun Modifier.placeholder(
+    visible: Boolean,
+    color: Color = PlaceholderDefaults.PlaceholderColor,
+    shape: Shape = RectangleShape
+) = takeIf { visible.not() } ?: composed(
+    inspectorInfo = debugInspectorInfo {
+        name = "placeholder"
+        value = visible
+        properties["visible"] = visible
+        properties["color"] = color
+        properties["shape"] = shape
+    }
+) {
+    PlaceholderModifier(
+        color = color,
+        shape = shape
+    )
+}
+
+/**
+ * If [visible], draw [shape] with [animatedBrush] instead of content.
+ *
+ * @sample com.google.accompanist.sample.placeholder.PlaceholderFadeSample
+ * @sample com.google.accompanist.sample.placeholder.PlaceholderShimmerSample
+ *
+ * @param visible defines whether the placeholder should be visible
+ * @param animatedBrush animated brush to paint placeholder with
+ * @param shape desired shape of the placeholder
+ */
+fun Modifier.placeholder(
+    visible: Boolean,
+    animatedBrush: PlaceholderAnimatedBrush,
+    shape: Shape = RectangleShape
+) = takeIf { visible.not() } ?: composed(
+    inspectorInfo = debugInspectorInfo {
+        name = "placeholder"
+        value = visible
+        properties["visible"] = visible
+        properties["animatedBrush"] = animatedBrush
+        properties["shape"] = shape
+    }
+) {
+    val infiniteTransition = rememberInfiniteTransition()
+    val progress by infiniteTransition.animateFloat(
+        initialValue = animatedBrush.initialValue(),
+        targetValue = animatedBrush.targetValue(),
+        animationSpec = animatedBrush.animationSpec()
+    )
+    PlaceholderModifier(
```

Comment:
This will create a new modifier on every animation frame. It's better if we make the brush `@Stable` and remember it. We can  then expose a `var progress by mutableStateOf(0f)` in the brush, and update it here using `brush.progress = progress`.

It results in the same thing but we only create one brush which is much better for performance.

Reply:
I was thinking about this too.
Thanks for the good suggestion!
I'll fix it with your suggestion.